src/lib/stack-client.js

```diff
@@ -71,6 +83,64 @@ const updateAccessToken = function(token) {
   throw new Error("updateAccessToken should not be used with a cozy-client instance initialization")
 }
 
+/**
+ * Fetch a resource with cozy-client
+ *
+ * Utility to maintain the compatibility with the legacy 
+ * standalone cozy-bar client
+ *
+ * @function
+ * @private
+ * @returns {Promise} the full raw JSON playload
+ */
+const fetchJSON = function(method, path, body, options={}) {
```

Comment:
This function should be async so that even if there is a throw in the code building the options, the function still returns a Promise (rejected) and not a synchronous error.